### PR TITLE
RMET-440 FileViewer Plugin - As an Android developer I can open a document

### PR DESCRIPTION
**Pull request by alexgerardojacinto**
## Description

Added native Android implementation for openDocument methods. Also added a method to open the file chooser.

## Context

Closes: https://outsystemsrd.atlassian.net/browse/RMET-440

## Type of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [x] Android
- [ ] iOS
- [ ] JavaScript

## Tests

Tested implemented methods locally. 
TODO: implement unit tests.

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [x] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java [hunk 1]
```diff
@@ -1,38 +1,205 @@
...
+            try {
+                this.cordova.getActivity().startActivity(intent);
+                callbackContext.success();
+            } catch (ActivityNotFoundException e) {
+                callbackContext.error(buildErrorResponse(5, "There is no app to open this document"));
+            }
+            return;
+        }
+
+        callbackContext.error(buildErrorResponse(2, "Path of the file to open is invalid"));
```
Comment (GFOutsystems) [inline src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java:122]: To achieve a good separation of concerns, all this part should be moved a new class that is independent from Cordova.

File: src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java [hunk 2]
```diff
@@ -1,38 +1,205 @@
...
+            try {
+                this.cordova.getActivity().startActivity(intent);
+                callbackContext.success();
+            } catch (ActivityNotFoundException e) {
+                callbackContext.error(buildErrorResponse(5, "There is no app to open this document"));
+            }
+            return;
+        }
+
+        callbackContext.error(buildErrorResponse(2, "URL to open is invalid"));
```
Comment (GFOutsystems) [inline src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java:151]: Same

File: src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java [hunk 3]
```diff
@@ -1,38 +1,205 @@
...
+        Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT);
+        intent.setType("*/*");
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        try {
+            this.cordova.getActivity().startActivityForResult(
+                    Intent.createChooser(intent, "Select a file to open"),
+                    0);
+        } catch (ActivityNotFoundException ex) {
+            callbackContext.error(buildErrorResponse(6, "There is no app to browse files with"));
         }
```
Comment (GFOutsystems) [inline src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java:173]: Same

File: src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java (hunk 1 above)
Comment (alexgerardojacinto) [reply src/android/com/outsystems/plugins/fileviewer/OSFileViewer.java:122]: Good point, on it!

**Review by GFOutsystems — APPROVED** [review]

**Review by andregrillo — APPROVED** [review]

**Pull request merged**